lib/web3/filter.js

```diff
@@ -149,6 +150,13 @@ var Filter = function (options, methods, formatter, callback) {
         } else {
             self.filterId = id;
 
+            // check if there are get pending callbacks as a consequence
+            // of calling get() with filterId unassigned.
+            self.getPendingCallbacks.forEach(function (cb){
```

Comment:
Hi, no problem with the renaming ;)

Yes I used it for a few days and worked, however I started to notice that sometimes not all logs were retrieved. Sometimes get() returned all the logs, sometimes one was missing, another times a few and in a pretty erratic pattern. To make things worse, the missing logs were not always the same :(.

The only explanation I can find is that this.implementation.getLogs(this.filterId, function(err, res) is not properly working.

I'll try to upload a test this afternoon/night to check that the asynchronous call of get is working and try to replicate the missing logs behaviour.